pkg/deployment/deployment.go

```
@@ -265,24 +268,36 @@ func ConditionalDeploy(
 				readyErrorMessage,
 				err.Error()))
 			return err
-		} else if ansibleEEJob.Status.Succeeded > 0 {
+		}
+
+		if ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusSucceeded {
 			log.Info(fmt.Sprintf("Condition %s ready", readyCondition))
 			status.Conditions.Set(condition.TrueCondition(
 				readyCondition,
 				readyMessage))
-		} else if ansibleEEJob.Status.Failed > 0 {
+			return nil
+		}
+
+		if ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusRunning || ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusPending {
+			log.Info(fmt.Sprintf("AnsibleEE job is still running: Execution: %s, Status: %s", ansibleEE.Name, ansibleEE.Status.JobStatus))
+			status.Conditions.Set(condition.FalseCondition(
+				readyCondition,
+				condition.RequestedReason,
+				condition.SeverityInfo,
+				readyWaitingMessage))
+			return err
+		}
+
+		if ansibleEE.Status.JobStatus == ansibleeev1alpha1.JobStatusFailed {
 			log.Info(fmt.Sprintf("Condition %s error", readyCondition))
-			err = fmt.Errorf("failed: job.name %s job.namespace %s", ansibleEEJob.Name, ansibleEEJob.Namespace)
+			err = fmt.Errorf("Execution.name %s Execution.namespace %s Execution.status.jobstatus: %s", ansibleEE.Name, ansibleEE.Namespace, ansibleEE.Status.JobStatus)
 			status.Conditions.Set(condition.FalseCondition(
 				readyCondition,
 				condition.ErrorReason,
 				condition.SeverityError,
 				readyErrorMessage,
 				err.Error()))
 			return err
```

ditto

---

Ignore this. We should return an error defined above :-(